improver/synthetic_data/set_up_test_cubes.py

```diff
@@ -206,6 +212,84 @@ def construct_scalar_time_coords(time, time_bounds, frt):
     return coord_dims
 
 
+def _create_dimension_coord(
+    coord_array, data, i, coord_name, coord_units, attributes=None
+):
```

Comment:
I'd probably simplify the definition a bit as:
```
_create_dimension_coord(coord_array, expected_len, **kwargs)
```
where the `**kwargs` match what's passed into `iris.coord.DimCoord`.  So you'd expect to use either standard_name or long_name, units and attributes (or not) and you can pass them straight through without having to define them here.

`expected_len` is just defined by `len(data[i])` and can be passed in from the top level.

Reply:
Updated [here](https://github.com/metoppv/improver/pull/1309/commits/1d7f21003ca70fbe79b8c8943e280bc37f411b98#diff-ea38258437543d699495d248fce61e29L214).

I left `coord_name` outside of the `**kwargs` as it's used in an error message. Also, `data_shape` is passed in as `data.shape` rather than the whole data array (also required for error message).

Reply:
Thanks for these changes.  I'm not sure I'd justify passing in these arguments for the error message, but I'm not going to insist on changing it - I think this looks good for the moment.